definitions/npm/a11y-dialog_v5.x.x/test_a11y-dialog_v5.x.x.js

```diff
@@ -0,0 +1,179 @@
+    // $ExpectError
+    dialog.create(true);
+  });
+});
+
+describe("The `show` method", () => {
+  it("should validate on default usage", () => {
+    dialog.show();
+  });
+
+  it("should validate when passing the `event` parameter", () => {
+    document.addEventListener("click", (evt: Event) => {
+      dialog.show(evt);
+    });
+  });
+
+  it("should error when passing erroneous parameters", () => {
+    // $ExpectError
+    dialog.show("");
+  });
+});
+
+describe("The `hide` method", () => {
+  it("should validate on default usage", () => {
+    dialog.hide();
+  });
+
+  it("should validate when passing the `event` parameter", () => {
+    document.addEventListener("click", (evt: Event) => {
+      dialog.hide(evt);
+    });
+  });
+
+  it("should error when passing erroneous parameters", () => {
+    // $ExpectError
+    dialog.hide("");
+  });
+});
+
+describe("The `destroy` method", () => {
+  it("should validate on default usage", () => {
+    dialog.destroy();
+  });
+
+  it("should error when passing erroneous parameters", () => {
+    // $ExpectError
+    dialog.create(true);
+  });
+});
+
+describe("The `on` method", () => {
+  it("should validate on default usage", () => {
+    dialog.on("create", () => {});
+    dialog.on("show", () => {});
+    dialog.on("hide", () => {});
+    dialog.on("destroy", () => {});
+  });
+
+  it("should validate on default usage with parameters", () => {
+    dialog.on("show", dialogEl => {
+      (dialogEl: HTMLElement);
+    });
+
+    dialog.on("show", (dialogEl, event) => {
+      if (event instanceof Event) {
+        (event: Event);
+      }
+    });
+  });
+
+  it("should error when passing erroneous parameters", () => {
+    // $ExpectError
+    dialog.on(() => {});
+    // $ExpectError
+    dialog.on("");
+    // $ExpectError
+    dialog.on("foobar");
+    // $ExpectError
+    dialog.on("show", true);
+  });
+
+  it("should error on missing parameters", () => {
+    // $ExpectError
+    dialog.on("show");
+  });
+});
+
+describe("The `off` method", () => {
+  it("should validate on default usage", () => {
+    dialog.off("create", () => {});
+    dialog.off("show", () => {});
+    dialog.off("hide", () => {});
+    dialog.off("destroy", () => {});
+  });
+
+  it("should validate on default usage with parameters", () => {
+    dialog.off("show", dialogEl => {
+      (dialogEl: HTMLElement);
+    });
+
+    dialog.off("show", (dialogEl, event) => {
+      if (event instanceof Event) {
+        (event: Event);
+      }
+    });
+  });
+
+  it("should error when passing erroneous parameters", () => {
+    // $ExpectError
+    dialog.off(() => {});
+    // $ExpectError
+    dialog.off("");
+    // $ExpectError
+    dialog.off("foobar");
+    // $ExpectError
+    dialog.off("show", true);
+  });
+
+  it("should error on missing parameters", () => {
+    // $ExpectError
+    dialog.off("show");
+  });
+});
+
+// describe("The `dialog` property", () => {
```

Comment:
Why this is commented out? I would rather remove it if it's not needed, or if it's something like "should work but can't make it work because of X", i would comment that there too.

Reply:
I need to give it another try. Open to suggestions. 
`dialog.dialog` can be either `HTMLDialogElement | HTMLElement`. 
Here I want to test only the `HTMLDialogElement` case, but I can't make Flow understand the condition.